FRAuth/FRAuth/WebAuthn/Authenticator/PlatformAuthenticator/PlatformAuthenticatorGetAssertionSession.swift

```diff
@@ -357,3 +358,21 @@ class PlatformAuthenticatorGetAssertionSession: AuthenticatorGetAssertionSession
         }
     }
 }
+
+// The key we are sorting will look like "22ba8202-6ece-4f56-9667-4eac6c265a41 20220621 20:10:10"
+extension Array where Element == String {
+    func sortByDate() -> [String] {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyyMMdd HH:mm:ss"
+        return self.map { (key) -> (String, Date?) in
+            let value: [String] = key.components(separatedBy: " ")
+            return (key, value.count > 1 ? formatter.date(from: value[1]) : nil)
+        }.sorted { lhs, rhs in
+            if let lhsDate = lhs.1 , let rhsDate = rhs.1 {
+                return lhsDate > rhsDate
+            }
+            return lhs.0 > rhs.0
+        }.map { $0.0 }
+    }
+}
```

Comment:
I think the name should be **sortedByDate** as it is a not-mutating method.

I'm suggesting this version of the method that is more readable and doesn't use the extra maps:

```
func sortedByDate() -> [String] {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyyMMdd HH:mm:ss"
        
        return self.sorted { string1, string2 in
            let stringComponents1 = string1.components(separatedBy: " ")
            let stringComponents2 = string2.components(separatedBy: " ")
            
            let date1 = stringComponents1.count > 1 ? formatter.date(from: stringComponents1[1]) : nil
            let date2 = stringComponents2.count > 1 ? formatter.date(from: stringComponents2[1]) : nil
            
            if let date1 = date1, let date2 = date2 {
                return date1 > date2
            }
            return string1 > string2
        }
    }
```

Reply:
With Map

.periyasamy/Projects/iOS/forgerock-ios-sdk/FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/WebAuthn/WebAuthnKeySelectorTests.swift:55: Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' measured [Time, seconds] average: 0.000, relative standard deviation: 173.540%, values: [0.002351, 0.000192, 0.000165, 0.000157, 0.000154, 0.000152, 0.000151, 0.000161, 0.000151, 0.000154], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' passed (0.275 seconds).



Without Map

/Users/jey.periyasamy/Projects/iOS/forgerock-ios-sdk/FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/WebAuthn/WebAuthnKeySelectorTests.swift:55: Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' measured [Time, seconds] average: 0.001, relative standard deviation: 170.804%, values: [0.005094, 0.000724, 0.000600, 0.000497, 0.000245, 0.000242, 0.000242, 0.000242, 0.000241, 0.000241], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100

Reply:
Both are closer for the average standard deviation . but i like vahan approach for the space complexity